__mocks__/windowMock.ts

```diff
@@ -19,14 +19,17 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-Object.defineProperty(window, "requestIdleCallback", {
-  writable: true,
-  value: jest.fn().mockImplementation(callback => callback()),
-});
-
-Object.defineProperty(window, "cancelIdleCallback", {
-  writable: true,
-  value: jest.fn(),
-});
+/**
+ * Mock the global window variable
+ */
+export function mockWindow() {
+  Object.defineProperty(window, "requestIdleCallback", {
+    writable: true,
+    value: jest.fn().mockImplementation(callback => callback()),
+  });
 
-export default {};
+  Object.defineProperty(window, "cancelIdleCallback", {
+    writable: true,
+    value: jest.fn(),
+  });
+}
```

Comment:
@Nokel81 Thanks for solving this!

Reply:
You are welcome! Hopefully the rest of this PR can be reviewed.